### ISSUE TITLE
Please be clearer about the meaning of the <MigrationTimestamp>_<MigrationName>.Designer.cs file

### ISSUE THREAD

**Issue by jimmcslim** [aspnet/EntityFramework.Docs#1868]
While this page describes the different files involved in an EF Core migration, it could be clearer about each of their roles. For example, what is the difference between the &lt;ContextName&gt;Snapshot.cs file and each &lt;MigrationTimestamp&gt;_&lt;MigrationName&gt;.Designer.cs file? What does EF Core use each of these files for? Convention suggests that we should keep our hands off the Designer.cs file, whereas it isn't as clear as to whether we can safely edit the Snapshot.cs file (although I don't). 

---
#### Document Details

⚠ *Do not edit this section. It is required for docs.microsoft.com ➟ GitHub issue linking.*

* ID: 6e28b0a8-b7e4-4c07-9e60-0df403bbee28
* Version Independent ID: 12300b31-6f7b-b75c-970d-693365e77013
* Content: [Migrations - EF Core](https://docs.microsoft.com/en-us/ef/core/managing-schemas/migrations/)
* Content Source: [entity-framework/core/managing-schemas/migrations/index.md](https://github.com/aspnet/EntityFramework.Docs/blob/master/entity-framework/core/managing-schemas/migrations/index.md)
* Product: **entity-framework**
* Technology: **entity-framework-core**
* GitHub Login: @bricelam
* Microsoft Alias: **bricelam**

**Comment by bricelam** [comment]
Designer file: https://stackoverflow.com/a/47060795/475031
Snapshot: https://docs.microsoft.com/en-us/ef/core/managing-schemas/migrations/teams

**Comment by ravindUwU** [comment]
Agreed! https://learn.microsoft.com/en-us/ef/core/managing-schemas/migrations/managing describes these as "metadata files that contain information used by EF", which is kinda vague and unhelpful 😬.

I think some _user-oriented_ documentation that describes what the designer files are, their purpose, implications of removing them (on the ability to roll back migrations, for example), etc. might be useful.

---

At a quick glance, both the DB snapshot & the migration designer files seem to use the same `Microsoft.EntityFrameworkCore.ModelBuilder` APIs to capture a snapshot of the DB; except the former represents the state of the DB after all migrations are applied, and the latter, the state after the corresponding migration is applied.

Does the designer file contain any metadata that isn't available in the migration file? I see the primary key names, foreign key names, index names, in both 👀.
